editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -4752,19 +4752,21 @@ void CanvasItemEditor::_popup_callback(int p_op) {
 		} break;
 		case CLEAR_GUIDES: {
 
-			if (EditorNode::get_singleton()->get_edited_scene()->has_meta("_edit_horizontal_guides_") || EditorNode::get_singleton()->get_edited_scene()->has_meta("_edit_vertical_guides_")) {
+			Node *const root = EditorNode::get_singleton()->get_edited_scene();
```

Comment:
We usually write `const Node *root`.

Reply:
`const Node *root` makes `root` a pointer to constant, while `Node *const root` makes the pointer itself constant. They are different types.

Also, the `undo_redo->add_do_method(root, ...)` later won't allow a `Node *const` parameter.

I think it's an existing pattern in the codebase, allowing the compiler to optimize the constant local variable away. For example:

https://github.com/godotengine/godot/blob/fbfd56c2477f1c05077a13533d11467ad3753ee4/editor/script_editor_debugger.cpp#L1257-L1262

Reply:
Thanks for the clarification!